src/Symfony/Component/Yaml/Yaml.php

```diff
@@ -30,6 +30,7 @@ class Yaml
     const DUMP_MULTI_LINE_LITERAL_BLOCK = 128;
     const PARSE_CONSTANT = 256;
     const DUMP_EMPTY_ARRAY_AS_SEQUENCE = 1024;
+    const PARSE_KEYS_AS_STRING = 2048;
```

Comment:
@GuilhemN the YAML file is not invalid. It is incompatible with **PHP** datastructures. YAML allows *any* value as key of a mapping (including floats, boolean, sequences and mapping).
PHP only supports strings and integers (and it accepts floats but casts them as integer, which is loosing data).